web/server.js

```diff
@@ -158,7 +158,7 @@ app.use('/', (req, res) => {
       res.setLocale(lr[0]);
     }
     const { locale } = res;
-    const fullUrl = `https://www.electricitymap.org${req.originalUrl}`;
+    const fullUrl = `https://www.electricitymap.org${req.baseUrl + req.path}`;
```

Comment:
Can we keep the `lang` query parameter in canonical, like suggested by @corradio and @skovhus in the issue? :) 
Could make a new variable for this, so we keep `fullUrl` like this (although maybe it should also be renamed now that it's not the full url 😄)

Reply:
Noted, I will do add the lang parameter !

Reply:
Updated the branch